Source documentation for geo-mechanics models

GiudGiud opened the pull request:
Splitting up the doco we had in tex form into the .md form

GiudGiud committed:
Add doco for the mechanics kernels syntaxes
refs #21970

GiudGiud committed:
Add documentation for geomechanics materials from tex source
refs #21970

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2344208/) on a816e1a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/28189/site)

This comment will be updated on new commits.

GiudGiud committed:
Move SM geo-mechanics models figures to large_media

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit 2fe4657bc59b93eb8179d22fa1b3ba137361b6b3
+Subproject commit 91c3eb793ec9c49b4af57532c4d8efd95c5b6d12
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

GiudGiud commented:
The eps are small but can't be used in moose docs afaik. That s why I made a png and moved everything to large_media

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakInclinedPlaneStressUpdate.md:
```diff
@@ -1 +1,21 @@
+the inclined plane's normal aligns with the $z$ axis, the assumptions
+listed in [this section](CappedWeakPlaneStressUpdate.md#assumptions) must hold.
+
+See the theory manual  (at [solid_mechanics/doc/theory/capped_weak_plane.pdf](https://github.com/idaholab/moose/modules/solid_mechanics/doc/theory/capped_weak_plane.pdf))
```
Does this PDF exist? I thought the `.md` documentation serves as the combined theory and user manuals. The link for this PDF does not work in the compiled version of the docs. I think this can be removed.

GiudGiud replied:
fixing the link, it does exist

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneCosseratStressUpdate.md:
```diff
@@ -1 +1,37 @@
+  $E_{ijkl}=E_{klij}$.
+
+
+See the theory manual  (at [solid_mechanics/doc/theory/capped_weak_plane.pdf](https://github.com/idaholab/moose/modules/solid_mechanics/doc/theory/capped_weak_plane.pdf))
+for more explanation.
```
Same comment as before. This PDF does not exist and probably doesn't need to be referenced.

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,767 @@
+p = \sigma_{zz} \ \ \ \mbox{and}\ \ \ q = \sqrt{\sigma_{xz}^{2} +
+  \sigma_{yz}^{2}} \ .
+\label{eqn.defn.p.q}
+\end{equation}
```
This is not rendering correctly because I don't think `\mbox` is supported in the .md equations. It works just fine in .tex.

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,767 @@
+\begin{equation}
+g = \left\{
+\begin{array}{ll}
+g_{0} & \mbox{ if } f = f_{0} \\
+g_{1} & \mbox{ if } f = f_{1} \\
+g_{2} & \mbox{ if } f = f_{2}\ .
```
`\mbox` usage causing rendering issues.

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,767 @@
+$i_{0}$ and $i_{1}$.   It is assumed
+that
+
+\begin{eqnarray}
+C & = & C(i_{0}) \ , \\
+\phi & = & \phi(i_{0}) \ , \\
+\psi & = & \psi(i_{0}) \ , \\
+S_{T} & = & S_{T}(i_{1}) \ , \\
+S_{C} & = & S_{C}(i_{1}) \ .
+\end{eqnarray}
```
It looks like eqnarray is not supported in `.md`. I looked elsewhere in the repository and it is only used in one case unrelated to this PR: `CombinedScalarDamage.md`, where it also does not render properly.

GiudGiud replied:
ah that's unfortunate. I ll figure something out

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,767 @@
+\frac{\partial q}{\partial\sigma_{xz}} = \frac{\sigma_{xz}}{q} \ .
+\end{equation}
+
+This means that Eqn~(\ref{eqn.xz.flow}) can be re-written
```
There is a different format in .md for referencing equations. I think you can simply use `This means that [eqn.xz.flow]` to properly reference the equation.

gambka commented on modules/solid_mechanics/doc/content/source/materials/TensileStressUpdate.md:
```diff
@@ -1 +1,56 @@
+
+The return-map algorithm first rotates $\sigma$ from the physical
+frame to the
+principal-stress frame (where $\sigma = \mbox{diag}(\sigma_{I}, \sigma_{II},
```
Usage of `\mbox` causing rendering issues.

gambka reviewed (commented):
Most comments are related to how equations are referenced and the use of seemingly unsupported `\mbox` and `equationarray` in `.md` files.

GiudGiud commented on modules/solid_mechanics/doc/theory/cosserat.tex:
```diff
@@ -71,7 +71,7 @@ \chapter{A 2D primer to Cosserat elasticity}
 \begin{figure}[htb]
 \begin{center}
-\includegraphics[width=0.8\textwidth]{inf_cosserat.eps}
+\includegraphics[width=0.8\textwidth]{../../../../large_media/solid_mechanics/theory/inf_cosserat.eps}
```
the .eps file got moved to large_media, it can still be used

gambka replied:
The `.eps` may still be used but if the point of the PR is to move content from .tex to .md, why keep the .tex files around?

GiudGiud replied:
I did not port everything either. the .tex files have their own narrative

GiudGiud commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,767 @@
+The elastic strain is
+
+\begin{equation}
+\ep_{ij}^{\mathrm{elastic}} = \ep_{ij}^{\mathrm{elastic, old}} +
```
ok I ll fix those

GiudGiud committed:
Address review:
- update link to pdf
- remove eqnarray
- use \text instead of \mbox
- use ids and [] links instead of latex ref
- couple symbol fixes

GiudGiud commented on modules/solid_mechanics/doc/content/source/materials/MultiParameterPlasticityStressUpdate.md:
```diff
@@ -0,0 +1,219 @@
+# MultiParameterPlasticityStressUpdate
```
that's weird. any page in the folder appears, even if the object did not exist

GiudGiud replied:
ok this isd normal then.
It s a base class it s not part of the syntax

however it is listed in the source list, because that lists all the existing pages

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,777 @@
+where the last equality holds by assumption (see full list of
+assumptions below).  The $xz$ and $yz$ components are similar:
+
+!equation id=eqn.xz.flow
```
I believe you can still use the \begin{equation} \end{equation} syntax with `\label{eqn.xz.flow}`, just the reference in the text would change from `~\ref{eqn.xz.flow}` to `[eqn.xz.flow].

GiudGiud replied:
I ll try that.
it does not like the dots in id=eqn.xz.flow so that other route is not working

GiudGiud committed:
Revert labeling of equations

GiudGiud committed:
Try to fix split equations

gambka commented on modules/solid_mechanics/doc/content/source/materials/MultiParameterPlasticityStressUpdate.md:
```diff
@@ -0,0 +1,221 @@
+An assumption that is made is that
+$E_{ab}$ is independent of the stress parameters, $s_{a}$ and the
+internal variables.  In this case\footnote{Special precautions are
+  taken when the eigenvalues are equal, as described in {\tt RankTwoTensor}.}
```
```suggestion
  taken when the eigenvalues are equal, as described in ``RankTwoTensor``.}
```

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,791 @@
+$q=\sqrt{\text{$\frac{1}{2}$}(\sigma_{xz}+\sigma_{zx})^{2} +
+  \text{$\frac{1}{2}$}(\sigma_{yz}+\sigma_{zy})^{2}}$, however the
+symmetrization is deliberately not written in
+Eqn~[eqn.defn.p.q] and below so that the equations also hold for
```
These are now being captured in the text. However, the `Eqn~` is not required as it duplicates text:
<img width="91" alt="Screenshot 2024-08-22 at 10 07 30 AM" src="https://github.com/user-attachments/assets/0a04d190-0428-4839-a893-1ece1a831e01">

GiudGiud replied:
changed
ugh even the ~ does not render right

gambka commented on modules/solid_mechanics/doc/content/source/materials/CappedWeakPlaneStressUpdate.md:
```diff
@@ -1 +1,791 @@
+
+In standard elasticity, the stress tensor is symmetric, so an
+equivalent definition of $q$ is
+$q=\sqrt{\text{$\frac{1}{2}$}(\sigma_{xz}+\sigma_{zx})^{2} +
+  \text{$\frac{1}{2}$}(\sigma_{yz}+\sigma_{zy})^{2}}$, however the
```
```suggestion
$q=\sqrt{\frac{1}{2}(\sigma_{xz}+\sigma_{zx})^{2} +
  \frac{1}{2}(\sigma_{yz}+\sigma_{zy})^{2}}$, however the
```

gambka reviewed (commented):
The  remaining things are split -> array for the multiline equations and removing duplicate words when referencing equations and figures. I put a few other suggestions in my review comments.

GiudGiud commented on modules/solid_mechanics/doc/content/source/materials/CombinedScalarDamage.md:
```diff
@@ -15,10 +15,12 @@ This model is a scalar damage model in which the stress $\boldsymbol{\sigma}$ is
 
-\begin{eqnarray}
+\begin{equation}
+\begin{split}
```
it's rebuilding with my latest changes right now. let s see if it fixed it in 30 minutes

GiudGiud commented on modules/solid_mechanics/doc/content/source/materials/MultiParameterPlasticityStressUpdate.md:
```diff
@@ -0,0 +1,221 @@
+\begin{equation}
+  s_{a} = s_{a}^{\mathrm{trial}} - \ga E_{ab} \frac{\partial
+    g}{\partial s_{a}} \ ,
+  \label{eqn.flow.rules}
```
that s fine though, you only need to number equations that are labelled (otherwise nowhere is referring to them)

GiudGiud committed:
Fix double labelling of equations

GiudGiud committed:
Fixup equations to make split command work

gambka reviewed (approved):
It looks like everything is rendering properly and all links work appropriately.

GiudGiud commented:
Thanks a lot for the review @gambka !

rpodgorney reviewed (approved):
Don't know if I am the best person to review this, but should be OK to merge

GiudGiud commented:
@lindsayad for the final review

GiudGiud commented:
Thanks Rob!

grmnptr reviewed (approved):
Very nice work!
